pkg/controller/account/iam_test.go

```diff
@@ -93,7 +96,33 @@ func TestGetSTSCredentials(t *testing.T) {
 	assert.Equal(t, creds.Credentials.SessionToken, SessionToken)
 	assert.NoError(t, err)
 
-	// TODO Test AWS Failure
+	// Test AWS Failure
+	expectedErr := awserr.New("AccessDenied", "", nil)
+	mockAWSClient.EXPECT().AssumeRole(gomock.Any()).Return(
+		&sts.AssumeRoleOutput{
+			Credentials: &sts.Credentials{
+				AccessKeyId:     AccessKeyId,
+				Expiration:      Expiration,
+				SecretAccessKey: SecretAccessKey,
+				SessionToken:    SessionToken,
+			},
+		},
+		expectedErr,
+	).Times(100)
+
+	// retries took long, need to mock it out
+	old := DefaultSleepDelay
+	DefaultSleepDelay = 0 * time.Millisecond
+	creds, err = getSTSCredentials(
+		nullLogger,
+		mockAWSClient,
+		"",
+		"",
+		"",
+	)
+	DefaultSleepDelay = old
```

Comment:
`DefaultSleepDelay` looks like a local variable, what happens if you remove this line?

Reply:
These are not local variables, they are package scoped and defined [here](https://github.com/openshift/aws-account-operator/pull/624/files#diff-06877cbab8eaefc13650b26e9c0b26fbb438a30862eb2b9666e0224421f51129R48-R50). I didn't think of another way of injecting a change to the retry time outside of doing something like this - if you have something cleaner in mind please shout :)

Reply:
I still don't think you need to reset these values back, I'm going to pull these changes down and test my theory just in case.

Reply:
For isolated testing purposes, probably not, but I generally feel it's good practice for our tests to not carry state from one to the next.

Reply:
Moved these to `init()` func to avoid resetting values per-test